src/ngx-smart-modal/tests/components/ngx-smart-modal.component.spec.ts

```diff
@@ -302,6 +302,27 @@ describe('NgxSmartModalComponent', () => {
     app.open();
   });
 
+  it('should add data to the modal', fakeAsync(() => {
+    inject([NgxSmartModalService], () => {
+      const fixture = TestBed.createComponent(NgxSmartModalComponent);
+      const app = fixture.debugElement.componentInstance;
+      const obj = {
+        prop1: 'test',
+        prop2: true,
+        prop3: [{ a: 'a', b: 'b' }, { c: 'c', d: 'd' }],
+        prop4: 327652175423
+      };
+      app.identifier = 'myModal';
+
+      expect(app.hasData()).toBeFalsy();
+      app.setData(obj, true);
+      tick();
```

Comment:
tick ?

Reply:
Because of the `markForChanges()` no `tick()` means no wait for the next detection cycle.